cmd/agent/app/run.go

```diff
@@ -179,6 +180,9 @@ func StartAgent() error {
 
 	// Setup expvar server
 	var port = config.Datadog.GetString("expvar_port")
+	if config.Datadog.GetBool("telemetry.enabled") {
+		http.Handle("/telemetry", promhttp.Handler())
```

Comment:
I think it can be good to use a local `prometheus.Registry` and then use [`prometheus.HandlerFor()`](https://godoc.org/github.com/prometheus/client_golang/prometheus/promhttp#HandlerFor)(https://godoc.org/github.com/prometheus/client_golang/prometheus#NewRegistry) it will be easer for testing.


> First, in more complex programs, global state is often quite problematic. That's the reason why the metrics constructors in the prometheus package do not interact with the global prometheus.DefaultRegisterer on their own. You are free to use the Register or MustRegister functions to register them with the global prometheus.DefaultRegisterer, but you could as well choose a local Registerer (usually created with prometheus.NewRegistry, but there are other scenarios, e.g. testing).

Reply:
I agree that it's a solution but I don't think that it is necessary, especially because we want the `telemetry` pkg to be callable from all parts of the Agent: we would rely on a global (to the package) instance of a `Registry` in the `telemetry` package, thus having the same behavior as using the default one of the prom package. WDYT?

Reply:
I agree that it will require a big refactoring about how we initialise the different Agent components: 
* avoid "func init()"
* use dependency injection or add dependency parameters to the different components.

We had the problem in another project when we wanted to validate metrics generation. because it not easy to cleanup metrics from the registry. 
but I think in your case you are in a better position, since you have added a facade in front of the prometheus api so you can add some mocking feature here.

Reply:
Yup it'll be relatively easy to change this due to the abstraction, I'm just not sure on what we would win with this change. If it is about the testability:
- if we want to validate metrics generation, we'll just have to create a mock implementation of the telemetry pkg.
- if we want to validate metrics generation of the prometheus package: that's not in the Agent that it must be done, it must be tested in the lib (prometheus) package.
However, I agree that it would not cost much to have a "local" registry in the telemetry package instead of using the one made available by the prometheus lib, I'll send the change.

Reply:
Last commit will now use a custom prometheus registry.